### ISSUE TITLE
QGIS crashes when editing two legends of the same map

### ISSUE THREAD

**Issue by JASMein03M** [qgis/QGIS#56266]
## User Feedback

Second time when editing two legends of the same map that QGIS crashes.
I was just editing two legends of the same map in the print layout when QGIS suddenly freezes and then a few seconds after also crashes.

## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Current thread 0x00000de4 (most recent call first):
<no Python frame>
```

**Stack Trace**

```

QgsAction::capture :
QgsLayerTreeView::updateExpandedStateFromNode :
QgsLayerTreeView::modelRowsInserted :
QObject::qt_static_metacall :
QAbstractItemModel::endInsertRows :
QSortFilterProxyModel::insertRows :
QSortFilterProxyModel::sortRoleChanged :
QSortFilterProxyModel::qt_static_metacall :
QObject::qt_static_metacall :
QAbstractItemModel::endInsertRows :
QgsLayerTreeModel::addLegendToLayer :
QgsLayerTreeModel::refreshLayerLegend :
QgsLayoutLegendWidget::mMoveDownToolButton_clicked :
QObject::qt_static_metacall :
QAbstractButton::clicked :
QAbstractButton::click :
QAbstractButton::mouseReleaseEvent :
QToolButton::mouseReleaseEvent :
QWidget::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
```

**QGIS Info**
QGIS Version: 3.28.6-Firenze
QGIS code revision: 868c9fa03b
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.6.4
Running against GDAL: 3.6.4

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.19045

### Steps to reproduce the issue

1. Make any map with a legend
2. Open the print layout
3. Insert two legends of this map
4. Start editing those two legends
5. Crash?

### Versions

The current available LTR version of QGIS is 3.28.15. 
The latest available version of QGIS is 3.34.3. 
We are working on 3.35-Master. It will go into feature freeze on 2024-01-19 12:00:00 UTC and be released as 3.36 on 2024-02-23 12:00:00 UTC. 
Details on changes can be found on github at https://github.com/qgis/QGIS/commits/master. 

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by elpaso** [comment]
Can you try 3.34 (I think it's fixed there) ? 3.28 is EOL.

**Comment by JASMein03M** [comment]
3.28 is LTR not EOL.

**Comment by elpaso** [comment]
> 3.28 is LTR not EOL.

until 2024-02-23, not a long time to live.

**Comment by nicogodet** [comment]
Duplicates https://github.com/qgis/QGIS/issues/52268

**Comment by agiudiceandrea** [comment]
@JASMein03M, anyway, the fix has also been backported to the 3.28 branch. So it shouldn't occur also using QGIS 3.28, provided you actually use the latest currently available version which is 3.28.15.

**Comment by JASMein03M** [comment]
Ah, alright. Thank you for all the information!
